flink-tests/src/test/java/org/apache/flink/runtime/operators/lifecycle/TestJobExecutor.java

```diff
@@ -84,6 +89,28 @@ public TestJobExecutor stopWithSavepoint(TemporaryFolder folder, boolean withDra
         return new TestJobExecutor(steps);
     }
 
+    public TestJobExecutor sendCommand(
+            TestCommandQueue commandQueue,
+            TestCommand testCommand,
+            TestCommandRetention retention) {
+        steps.add(ctx -> commandQueue.add(testCommand, TestCommandTarget.ALL, retention));
+        return this;
+    }
+
+    public TestJobExecutor waitForTermination() {
+        steps.add(
+                ctx -> {
+                    while (!ctx.miniClusterResource
+                            .getClusterClient()
+                            .getJobStatus(ctx.job)
+                            .get()
+                            .isGloballyTerminalState()) {
+                        Thread.sleep(100);
```

Comment:
nit: `sleep(1)`? In the worst case `sleep(10)`?

Reply:
This will save 99ms at most, or am I missing something?
(I'll change it anyways)